src/armature/FillVolume.ts

```diff
@@ -0,0 +1,143 @@
+import { AABB } from '../geometry/BakedGeometry';
+import { worldSpaceAABB } from '../utils/aabb';
+import { Cost, GeneratorInstance, SpawnPoint } from './Generator';
+import { Model } from './Model';
+import { GeometryNode, Node } from './Node';
+
+import { vec4 } from 'gl-matrix';
+import { range } from 'lodash';
+
+type Grid = { [key: string]: true };
+
+/**
+ * Creates a cost function based on how much of a target volume a shape fills.
+ */
+export class FillVolume {
```

Comment:
Could we have an abstract base class or interface for `Cost`, with methods `getCost` here?

Reply:
I see that you have the `CostFn` interface, could you please specify that this class is implementing it?